### ISSUE TITLE
Adding a config allowing external images

### ISSUE THREAD

**Issue by zyuhel** [oscarotero/Embed#133]
I came on to an issue when image is not fetched. If main post image links to external site it won't be get. As i know it is a desired behavior, but may be we need a config option like get bigger image that can be tuned (instead of creating different providers for the sites that will have this). 

http://www.brothers-brick.com/2016/04/06/stunning-lego-darth-vader-mask-cleverly-hides-scenes-from-star-wars/
Also some wordpress blogs hosted on a custom domains still serve images from wp.com subdomains. Adding exception for it will be a bit dirty. so configurable option is pretty good variant.

**Comment by oscarotero** [comment]
This is to prevent to get banners or other similar images, but it's true that some external images are relevant. Maybe an option like this:

```php
$options = [
    'providers' => [
        'html' => [
            'externalImages' => true //by default is false
        ]
    ]
]
```

And even allow to provide url patterns:

```php
$options = [
    'providers' => [
        'html' => [
            'externalImages' => [
                '*.flickr.com/*'
            ]
        ]
    ]
]
```

**Comment by zyuhel** [comment]
:) may be know there is a need of external factory that will hold configurations for different urls :+1: 
Thank you.

**Comment by oscarotero** [comment]
A simple way:

```php
use Embed\Request;
use Embed\Embed;

$request = new Request('http://example.com');

if ($request->getDomain() === 'example') {
    $config = getExampleConfig();
} else {
    $config = getDefaultConfig();
}

$info = Embed::create($request, $config);
```

**Comment by oscarotero** [comment]
Closed because it's already implemented and will be available in the 2.7 release.
